### DISCUSSION TITLE
How to understand currency of currentPrice in OrderV2?

### DISCUSSION THREAD

**Discussion by metakoper** [ProjectOpenSea/opensea-js#1319] [Q&A]
How can I understand the currency of an OrdrerV2 currentPrice? 
```
openSeaSDK.api.getOrder({
            side: OrderSide.ASK,
            assetContractAddress: request.assetContractAddress,
            tokenIds: [request.tokenId],
            orderBy: "eth_price",
            orderDirection: "asc"
        })
```
Even though `orderBy` option is called `eth_price` it actually compares by usd price, and the `currentPrice` returned is not wei but depends on the actual currency of the order.

**Comment by ryanio** [comment]
getOrder is deprecated, please update to v7 and try getOrders, let us know if the sorting there is not working to your expectations

you can also call getBestListing() now

**Reply by metakoper** [reply]
Hi @ryanio 🙏 
I'm actually using v7.
`getOrders` - also returns `OrderV2` so there here's no currency / symbol in the price as well.
`getBestListing` - is not reliable, for many collections it returns
`GET https://api.opensea.io/v2/listings/collection/parallelalpha/nfts/10795/best`
```json
{
  errors: [
    'This endpoint only supports collections with a single asset contract'
  ]
}
```

**Reply by ryanio** [reply]
gotcha thanks, try `getBestListings()` that seems  to work with `parallelalpha`

**Reply by metakoper** [reply]
got you!
but there's another problem here 😅 (not speaking about iterating through all the listings as you only specify collection slug):
- it only returns 100 listings in response (700+ of the items in collection have listings)
- it **doesn't** return next cursor for some reason

**Reply by metakoper** [reply]
The problem is also that you **require** an `OrderV2` object in order to use `openSeaSDK.fulfillOrder`

**Reply by ryanio** [reply]
@metakoper if you want to iterate through all listings use `getAllListings()`

I'll open a PR to improve fulfillOrder that can accept a listing/offer response from the api (update: opened https://github.com/ProjectOpenSea/opensea-js/pull/1320)

**Reply by ryanio** [reply]
The PR is released as part of [v7.0.1](https://github.com/ProjectOpenSea/opensea-js/releases/tag/v7.0.1), give it a try and let me know if it works for you!
